spark/src/test/java/io/ddf/spark/SparkDDFManagerTests.java

```diff
@@ -125,9 +129,37 @@ public void testBasicCopyForS3(S3DDFManager s3DDFManager) throws DDFException {
     DDF pqtSparkDDF = manager.copyFrom(pqtDDF);
     LOG.info(pqtSparkDDF.sql("select * from @this limit 5", "error").getRows().toString());
     LOG.info("========== avro ==========");
-    //S3DDF avroDDF = s3DDFManager.newDDF("adatao-sample-data", "test/avro/partition/", null, null);
-    //DDF avroSparkDDF = manager.copyFrom(avroDDF);
-    //LOG.info(avroSparkDDF.sql("select * from @this limit 5", "error").getRows().toString());
+    S3DDF avroDDF = s3DDFManager.newDDF("adatao-sample-data", "test/avro/partition/", null, null);
+    DDF avroSparkDDF = manager.copyFrom(avroDDF);
+    assert (avroSparkDDF.getNumRows() > 0);
+
+
+    LOG.info("========== testFolder/folder ==========");
+    final List<String> nameList = new ImmutableList.Builder<String>()
+        .add("dd2_2") // valid
+        .add("_dd2_2") // start with dash
+        .add("2dd2_2") // start with num
+        .add("@dd2_2") // start with others
+        .add("dd 22") // space
+        .add("dd 22 ") // back space
+        .build();
+
+    final List<String> expectedList = new ImmutableList.Builder<String>()
+        .add("dd2_2")
+        .add("dd2_2_0")
+        .add("dd2_2_1")
+        .add("dd2_2_2")
+        .add("dd_22")
+        .add("dd_22_0")
+        .build();
+
+    StringBuilder schema = new StringBuilder(String.format("%s int", nameList.get(0)));
+    for (int i = 1; i < nameList.size(); ++i) {
+      schema.append(String.format(", %s int", nameList.get(i)));
+    }
+    S3DDF ddf = s3DDFManager.newDDF("adatao-sample-data", "test/csv/6col.csv", schema.toString(), null);
```

Comment:
Please help put this file under `adatao-sample-data/test/pa/<folder>/<filename>`, the reason for this is that other part of this `test` folder usually get deleted and upload again using a zip file, so if putting it under `csv`, this file may get deleted